### PR TITLE
Remove onnxruntime_session_options_config_keys.h from c_api

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Remove onnxruntime_session_options_config_keys.h from c_api

**Motivation and Context**
- We actually do not need the onnxruntime_session_options_config_keys.h in the c_api
- Add onnxruntime_session_options_config_keys.h to where it is used

### CODE REVIEW

File: tools/ci_build/github/linux/copy_strip_binary.sh [hunk 1]
```diff
@@ -36,7 +36,6 @@ cp $SOURCE_DIR/include/onnxruntime/core/session/onnxruntime_cxx_api.h  $BINARY_D
 cp $SOURCE_DIR/include/onnxruntime/core/session/onnxruntime_cxx_inline.h  $BINARY_DIR/$ARTIFACT_NAME/include
 cp $SOURCE_DIR/include/onnxruntime/core/providers/cpu/cpu_provider_factory.h  $BINARY_DIR/$ARTIFACT_NAME/include
 cp $SOURCE_DIR/include/onnxruntime/core/providers/cuda/cuda_provider_factory.h  $BINARY_DIR/$ARTIFACT_NAME/include
-cp $SOURCE_DIR/include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h  $BINARY_DIR/$ARTIFACT_NAME/include
```
Comment (skottmckay) [inline tools/ci_build/github/linux/copy_strip_binary.sh:39]: Do we still want it here so the values are discoverable from this package?

**Comment by pranavsharma** [comment]
API usability question: a developer will now have to find the config file in the source tree for a specific release branch to learn about the string constants. If it's part of the package he/she can immediately find all that's needed to use the API in one place. Can we still package the file as part of the shipped artifacts? Both Nuget and zip files.

File: tools/ci_build/github/linux/copy_strip_binary.sh (hunk 1 above)
Comment (gwang-msft) [reply tools/ci_build/github/linux/copy_strip_binary.sh:39]: Will keep the header file for the package

**Comment by gwang-msft** [comment]
> API usability question: a developer will now have to find the config file in the source tree for a specific release branch to learn about the string constants. If it's part of the package he/she can immediately find all that's needed to use the API in one place. Can we still package the file as part of the shipped artifacts? Both Nuget and zip files.

Added it back

**Review by skottmckay — APPROVED** [review]

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**